hal/src/stm32f2xx/sleep_hal.cpp

```diff
@@ -186,7 +309,36 @@ static int validateWakeupSource(hal_sleep_mode_t mode, const hal_wakeup_source_b
     return SYSTEM_ERROR_NOT_SUPPORTED;
 }
 
-static void enterStopSleep() {
+static void enterPlatformSleepMode() {
+    int basePri = __get_BASEPRI();
+    __set_BASEPRI(2 << (8 - __NVIC_PRIO_BITS));
```

Comment:
On second though, this will potentially break other wakeup sources, right? Unless we bump their priorities. Same with USART PR. Also, have we verified that we don't have any interrupts with priority 1 enabled by default?

Reply:
>this will potentially break other wakeup sources, right?

Good catch! I'll bump all wakeup sources' priority for the generic `__WFI()` mode.

>Also, have we verified that we don't have any interrupts with priority 1 enabled by default?

Good point! I dumped all the `NVIC->IP` and `NVIC->ISER` registers after entering the `hal_sleep_enter()` function and found that only the `RTC_WKUP_IRQn` is set with priority 1, but it is not enabled. So we can safely set `BASEPRI` to 2 and bump the priority to 1 for necessary wakeup sources.